src/lib/alloc.c

```diff
@@ -249,14 +259,15 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
 	}
 
 	/* we found enough space, let's allocate it */
-	start = map->first_free;
 	map->free_count -= count;
 	ptr = (void *)(map->base + start * map->block_size);
 	hdr = &map->block[start];
 	hdr->size = count;
 	heap->info.used += count * map->block_size;
 	heap->info.free -= count * map->block_size;
-	map->first_free = map->first_free + count;
+	/* update first_free if needed */
+	if (map->first_free == start)
```

Comment:
You probably wanted `if (map->first_free != start)`

Reply:
No, if (map->first_free != start), we don't need to update map->first_free.

Reply:
Yes, this is correct.